### ISSUE TITLE
[BUG] harvester-cloud-provider rely on CNI to communicate with host

### ISSUE THREAD

**Issue by FrankYang0529** [harvester/harvester#1999]
**Describe the bug**
`harvester-cloud-provider` needs to wait for CNI to be ready, so it can communicate with the host. However, `harvester-cloud-provider` only communicates with host to get some provider metadata, so we should use `hostNetwork: true`.

**To Reproduce**
1. Setup master-head harvester cluster.
2. Setup rancher v2.6.4-rc6.
3. Import harvester cluster to rancher.
4. Run `1.22.7+rke2r1` RKE2 on harvester.

**Expected behavior**
harvester-cloud-provider can communicate with harvester and remove the taint in RKE2.

**Support bundle**

Logs in harvester-cloud-provider.

```
node_controller.go:390] Initializing node rke2-in-jeos-pool1-51a1e43e-fm4kk with cloud provider
node_controller.go:212] error syncing 'rke2-in-jeos-pool1-51a1e43e-fm4kk': failed to get instance metadata for node rke2-in-jeos-pool1-51a1e43e-fm4kk: Get "https://192.168.0.131:6443/apis/kubevirt.io/v1/namespaces/default/virtualmachines/rke2-in-jeos-pool1-51a1e43e-fm4kk": dial tcp 192.168.0.131:6443: i/o timeout, requeuing
```

**Environment:**
 - Harvester ISO version: master-head
 - Rancher version: v2.6.4-rc6
 - Underlying Infrastructure: KVM

Related to: https://github.com/rancher/rancher/issues/36716

**Comment by FrankYang0529** [comment]
cc @guangbochen

**Comment by FrankYang0529** [comment]
Test `v1.22.7-rc2+rke2r2` on Focal and JeOS. Both can work.

Test steps:
* Setup Harvester `master-75cc3ced-head` and Rancher `v2.6-head`.
* Create RKE2 `1.22.7+rke2r1` cluster. (1.22.7+rke2r2 is not released yet)
* After clicking `Create` button, click `Edit YAML` and  change `kubernetesVersion` to `v1.22.7-rc2+rke2r2`.
* After k8s installed, ssh into the VM and delete harvester-cloud-provider image version v0.1.1 ReplicaSet.

![Screen Shot 2022-03-14 at 11 16 35 AM](https://user-images.githubusercontent.com/4927361/158098868-3c98e303-35ae-4d31-beea-e6bdcbde8858.png)

**Comment by FrankYang0529** [comment]
Test on Rancher `v2.6-head`(059b5c504f31). RKE2 `1.22.7+rke2r2` works fine with Focal and JeOS.

![Screen Shot 2022-03-16 at 1 54 55 PM](https://user-images.githubusercontent.com/4927361/158525607-3fc51b0d-74aa-4ac5-9ddb-0b720e7e89a3.png)

**Comment by TachunLin** [comment]
Verified **fixed** on Rancher `v2.6-head 2642acc` (03/16) with Harvester `v1.0-591d8015-head` (03/16). Close this issue.

### Result
1. Can provision RKE2 cluster with kubernetes version **`v1.22.7+rke2r2`**
  ![image](https://user-images.githubusercontent.com/29251855/158577116-0bda5f89-2083-49e9-8f55-d6d172df05ef.png)

2. Host network `enabled` in cloud provider yaml 
3. Harvester cloud provider image bumped to `v1.0.2`
  ![image](https://user-images.githubusercontent.com/29251855/158762920-a798fb5e-740d-4b75-bd3b-ff3d598ae535.png)
4. `Affinity` added to cloud provider pod
  ![image](https://user-images.githubusercontent.com/29251855/158763106-bfe64062-2e7d-4ef7-b131-671efbdf6800.png)

### Test Information
* Test Environment: 1 nodes harvester on local kvm machine
* Harvester version: v1.0-591d8015-head (03/16)
* Rancher version: v2.6-head 2642acc
* Kubernetes version: v1.22.7+rke2r2
* OS image: Ubuntu focal

### Verify Steps
1. Create a one node harvester 
2. Upload Ubuntu focal and SLES JeOS image
3. Create virtual network vlan1 with id 1
6. Import Harvester from Rancher
7. Create cloud credential 
8. Provision an RKE2 cluster 
9. Add the following use cloud-init data 
```
  password: 123456
  chpasswd: { expire: False }
  ssh_pwauth: True
  ```
10. Select Ubuntu focal image
11. Provide available cpu and memory resources
12. Provision the RKE2 cluster
13. After provisioning done
14. Open Workload -> Deployments -> harvester-cloud-provider 
15. Edit Yaml
